File: src/hb/rule/tag/void.c
#include <hb/collection.h>
#include <hb/rule.h>
#include <hb/rune.h>

static hb_set_tag_names* hb_rule_tag_void_set;

void hb_rule_tag_void_add_elems(hb_set_tag_names* set)
{
	hb_set_tag_names_add_whole_literal(set, "area");
	hb_set_tag_names_add_whole_literal(set, "base");
	hb_set_tag_names_add_whole_literal(set, "br");
	hb_set_tag_names_add_whole_literal(set, "col");
	hb_set_tag_names_add_whole_literal(set, "embed");
	hb_set_tag_names_add_whole_literal(set, "hr");
	hb_set_tag_names_add_whole_literal(set, "img");
	hb_set_tag_names_add_whole_literal(set, "input");
	hb_set_tag_names_add_whole_literal(set, "keygen");
	hb_set_tag_names_add_whole_literal(set, "link");
	hb_set_tag_names_add_whole_literal(set, "meta");
	hb_set_tag_names_add_whole_literal(set, "param");
	hb_set_tag_names_add_whole_literal(set, "source");
	hb_set_tag_names_add_whole_literal(set, "track");
	hb_set_tag_names_add_whole_literal(set, "wbr");
}

void hb_rule_tag_void_init(void)
{
	hb_rule_tag_void_set = hb_set_tag_names_create();
	hb_rule_tag_void_add_elems(hb_rule_tag_void_set);
}

bool hb_rule_tag_void_check(nh_view_str* tag)
{
	return hb_set_tag_names_has(hb_rule_tag_void_set, tag);
}

File: rust/common/spec/tag/void.rs
use lazy_static::lazy_static;
use rustc_hash::FxHashSet;

lazy_static! {
    pub static ref VOID_TAGS: FxHashSet<&'static [u8]> = {
        let mut s = FxHashSet::<&'static [u8]>::default();
        s.insert(b"area");
        s.insert(b"base");
        s.insert(b"br");
        s.insert(b"col");
        s.insert(b"embed");
        s.insert(b"hr");
        s.insert(b"img");
        s.insert(b"input");
        s.insert(b"keygen");
        s.insert(b"link");
        s.insert(b"meta");
        s.insert(b"param");
        s.insert(b"source");
        s.insert(b"track");
        s.insert(b"wbr");
        s
    };
}
